### DISCUSSION TITLE
Issues saving/annotating downloaded PDF files with equations

### DISCUSSION THREAD

**Discussion by ryoungblom** [knowledgefutures/pubpub#1288] [Support]
Hello, all!

I'm having kind of an odd issue that I haven't quite pinned down yet, but that I think I have enough info to report.  I've noticed an issue with annotating PDF files that are downloaded from PubPub, but only those with LaTeX equations. It happened with my own pubs, and I thought it was maybe just an issue with my organization somehow, until I noticed that ONE of my pubs worked, and it just happened to be a more narrative piece with no equations.  I did some very-small-sample-size testing, and I feel like the issue arises specifically with pubs that have equations in them.  This issue also seems limited to PDF viewers that are NOT Preview (the MacOS built-in PDF reader). Personally, I'm using Acrobat for this.

So, for example, take these two pubs:
1) https://baas.aas.org/pub/2020i0330/release/1
2) https://tmb.apaopen.org/pub/daily-stress-and-affect/release/1

(just randomly selected as the first two from different organizations that I found with equations)

And compare against these two:
3) https://baas.aas.org/pub/2009p913/release/1
4) https://tmb.apaopen.org/pub/9bncjd33/release/8

(just the first two pubs I found from these same organizations that didn't seem to have equations)

When downloading the first two (numbers 1 and 2), I am able to download them, open them, and view them, equations included.  Same with the second two (numbers 3 and 4).  The issue arises when I try to annotate the PDFs that have equations with highlighting, etc.  When I open numbers 3 and 4, I can highlight no problem, and then I can save the copies no problem.  When they're re-opened, the highlighting is still there, indicating a valid save.

However, when annotating numbers 1 and 2, I can open, read, and highlight, but I am unable to actually save the file (which means I am unable to preserve my annotations).  There is some kind of error that prevents exporting files with equations in them.  This seems to be something that happening inside PubPub, as other pdf files with LaTeX from other sources (i.e., just picking a random paper off arxiv: https://arxiv.org/pdf/2102.07249.pdf) highlight and save just fine.

The issue seems to appear on a few different PDF programs, but as I said, I'm mostly working on Acrobat Reader. I have tried Mac and Windows versions of Acrobat, and the issue also occurs when editing PDFs in Edge (on Windows). Preview (Mac) seems to be able to open and annotate and save, but about half of the highlights disappear when reopening the file (but this is almost definitely just be a Preview issue, as Preview has a lot of trouble with annotations recently).

Any idea what might be causing this problem? Please keep in mind, I'm not 100% sure about the equations component being the key, that's just the only consistency that I've noticed so far.  Any other insights, even those disproving the equations hypothesis, would be very appreciated. Thanks!

**Steps to reproduce:**
1) Find a PubPub pub with equations (there are a few listed above)
2) download PDF
3) open PDF in either Acrobat or Edge on either Mac (Acrobat) or Windows (Acrobat/Edge)
4) Add an annotation (highlighting)
5) Try to save file with annotation

_Expected Behavior_: File saves
_Observed Behavior_: Error message (the error message seems to vary by PDF)

**OPTIONAL:**
6) find a PubPub pub without equations (there are a few listed above)
7) download PDF
8) open PDF in either Acrobat or Edge
9) add an annotation (highlighting)
10) Try to save file with annotation

_Expected Behavior_: File saves
_Observed behavior_: File saves

**Comment by ryoungblom** [comment]
***Updating to add:  I'm leaving my original post/theory for posterity, but Gabe (a PubPub maintainer) does NOT think that metadata is the issue (see Gabe's comment, below).  However, for whatever reason, the workaround still works, so if you're having this issue, definitely try the workaround***

Additional Note + a workaround for anyone else who runs into this issue:  I wonder if there's an element of Metadata being an issue here?  I found that if I take a PDF with equations downloaded straight from PubPub and run it through a metadata editing platform (i.e., I've been using pdfcandy) to edit the metadata, I can clear the "title" field (which is the only field that I can see as filled-in on pdfcandy) and then the PDF behaves as expected.

I have no idea how pdfcandy works behind the scenes—it only allows one to edit 6 fields, so a few things could be happening here:

1) it could be the case that there are a lot of metadata fields that are filled with data on these PDFs from PubPub, but you can only edit six on pdfcandy. So, when you edit the available fields on pdfcandy, it then **erases all other metadata** (implying that blank metadata fields fix this); or
2) it could be the case that there are a lot of metadata fields that are filled with data on these PDFs from PubPub, and when you edit the available fields on pdfcandy, it **does not erase any other metadata** (implying that the title field is the only issue, since there would theoretically still be PubPub-generated metadata in the PDF file); or
3) it could be the case that the fields I can see on pdfcandy are the only available fields (i.e., there's no additional metadata attached to this PDF), in which case, it's possible that either a) the title field is the issue, and other metadata wouldn't hurt things; or b) it actually just needs blank metadata, so removing the title _happens_ to be the fix because removing the title clears all of the metadata.

I will keep experimenting and see if I can figure out any more details.

**For anyone else who runs into this problem**:  So far, the fix that works for me is downloading the PDF right from PubPub, putting it into pdfcandy to edit the metadata, and clearing the title field.  I assume that clearing any fields that populate with information would be the best strategy, if your PDF has more than just title metadata (but this is definitely a valid experiment that I have not done). I also assume that any other metadata-editing method would work. I believe you can do this with Acrobat, if you have the Pro version, and this may be easier than uploading into the cloud.

**Reply by gabestein** [reply]
Thanks so much for reporting and experimenting with this issue. I took a look at it the other day and, well, it's a weird one.

I don't know exactly what's going on, but I suspect it has less to do with metadata, and more with an incompatibility in how we're generating the PDFs when math is included (maybe something to do with character sets?). I was able to re-export the BaaS PDF using Apple Preview with title metadata, and successfully save annotations in Acrobat. My guess is that this experiment, and your pdfcandy experiment, work because they use more compatible libraries to generate their PDFs.

We'll take a closer look in the coming days and see what our options are.

**Reply by ryoungblom** [reply]
No worries! It's a strange one, for sure. So far, the workaround is still solid for me, so hopefully that gets anyone else who runs into this issue through until there's a good fix :)

Interesting that you got Preview to work for you... if that works for most people, that would definitely be a better workaround than pdfcandy.  The reason I didn't suggest this as a fix is that, on my machine, I can open the pdf in Preview, and export it, but I lose some annotating ability.  I still haven't figured out the exact parameters around what's  happening, but as far as I can tell, I lose the ability to highlight anything that's in "paragraph" text on PubPub.

So, for example, I can download the pdf, open in Preview, export as PDF, and open in Acrobat just fine. However, while a freshly-downloaded version (or a version that's been run through pdfcandy) will let me select literally any section of text I want (and selecting is a prerequisite for highlighting), pdfs that have been downloaded from PubPub and the exported as PDF by Preview lose the ability to select "normal" or "paragraph" text.  My cursor just glides over it like it's an image... I can't select anything, I can't get the little "hover over text" cursor (no idea what it's called, the little "seriffed capital I" shaped cursor), and I can't select.

I can still select anything that's NOT in paragraph text (titles, headings, footnotes, etc.), and I can highlight those and save.  I can also leave notes.  I just can't select or highlight pdfs that have been downloaded from PubPub and then exported via Preview.

If it's working for you, it might just be my machine... I'll test on other machines this weekend.

Thanks!

**Reply by gabestein** [reply]
Ah, actually, I'm seeing that, too for paragraph text. I had only checked on the cover page. Even more strangely, I can highlight in Preview and that shows up and can be deleted in Acrobat. But I can't create new paragraph highlights on Acrobat. Super weird.

**Reply by gabestein** [reply]
Did a little more investigating. I think this may be a relatively newly introduced bug in Acrobat (see: https://community.adobe.com/t5/acrobat-reader/help-showing-error-message-135-when-trying-to-save-as/td-p/11221157).

We'll keep taking a look. But in the meantime, I believe we were able to solve the issue for Preview, which was related to an incompatible font, which should provide a slightly better workaround than re-exporting the PDF. This will only work when PDFs are regenerated, which happens on a new release or when changes to a draft are made. But if there are specific PDFs you'd like us to clear the cache for, we can.

**Reply by ryoungblom** [reply]
Thanks for sticking with this!  I had a chance to test the fix that allows for exporting from Preview, and that worked well for me.  I was able to open and export in Preview, and then annotate in Acrobat. No issues so far, which makes my life quite a bit easier.... this is way better than using a third-party website.

I'll have to test on Windows to see what options there are for exporting and then annotating in Acrobat, but for anyone who is looking for a solution on Mac, the best workaround so far seems to be:

1) Download PDF from PubPub
2) Open PDF in Preview
3) Use the File -> Export as PDF option to create a new PDF
4) Open new PDF in Acrobat—it should be able to be annotated and saved in Acrobat!

Will test out Windows options this week--

Thanks!
